### PR TITLE
bugFix/Prevent PIN bruteForce

### PR DESCRIPTION
**Pull request by alazarevski**
Pin Code Entry UI bruteforce prevention

### CODE REVIEW

File: src/actions/appActions.js [hunk 1]
```diff
@@ -148,6 +148,15 @@ export const initAppAndRedirectAction = (appState: string, platform: string) =>
       const { connectedProviders = [] } = await storage.get('exchangeProviders');
       dispatch({ type: SET_CONNECTED_EXCHANGE_PROVIDERS, payload: connectedProviders });
 
+      const { pinAttemptsCount = 0, lastPinAttempt = 0 } = wallet;
```
Comment (jsidorenko) [inline src/actions/appActions.js]: How about putting this into app_settings?

File: src/screens/PinCodeUnlock/PinCodeUnlock.js [hunk 2]
```diff
@@ -93,9 +106,44 @@ class PinCodeUnlock extends React.Component<Props> {
     }
   }
 
+  getWaitingTime = (onMount) => {
+    const { pinAttemptsCount, lastPinAttempt } = this.props.wallet;
+    const lastAttemptSeconds = (Date.now() - lastPinAttempt) / 1000;
+    const nextInterval = pinAttemptsCount * 2;
+    if (pinAttemptsCount > 5) {
```
Comment (jsidorenko) [inline src/screens/PinCodeUnlock/PinCodeUnlock.js]: Let's extract this into config

File: src/screens/PinCodeUnlock/PinCodeUnlock.js [hunk 3]
```diff
@@ -93,9 +106,44 @@ class PinCodeUnlock extends React.Component<Props> {
...
+  handleLocking = (onMount) => {
+    if (this.interval) {
+      clearInterval(this.interval);
+    }
+    const waitingTime = this.getWaitingTime(onMount);
+    if (waitingTime > 0) {
+      this.setState({ waitingTime }, () => {
+        this.interval = setInterval(() => {
+          if (this.state.waitingTime > 0) {
+            this.setState({ waitingTime: this.state.waitingTime - 1 });
```
Comment (jsidorenko) [inline src/screens/PinCodeUnlock/PinCodeUnlock.js]:
```suggestion
            this.setState((prev: State) => ({ waitingTime: prev.state.waitingTime - 1 }));
```

File: src/screens/PinCodeUnlock/PinCodeUnlock.js [hunk 4]
```diff
@@ -93,9 +106,44 @@ class PinCodeUnlock extends React.Component<Props> {
     }
   }
 
+  getWaitingTime = (onMount) => {
```
Comment (jsidorenko) [inline src/screens/PinCodeUnlock/PinCodeUnlock.js]: according to the name, `onMount` should be a function, but here it's a boolean

[the remark this answers is not in the document]
Comment (alazarevski) [reply src/screens/PinCodeUnlock/PinCodeUnlock.js]: 8570819

**Pull request merged**